KitapYurdu.com.js

```diff
@@ -0,0 +1,303 @@
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+/*
+Known issues:
+- Turkish char problem: using ZU.capitalizeTitle with publisher names with "I" returns "i" instead of "ı".
+*/
+
+function detectWeb(doc, url) {
+	if (url.indexOf('/kitap/')>-1||url.indexOf('product_id')>-1) {
+		return 'book';
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//a[contains(@href, "/kitap/")]|//a[contains(@href, "product_id")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+/*
+remove titles from creators
+*/
+
+function cleanCreatorTitles(str){
+	return str.replace(/Prof.|Doç.|Yrd.|Dr.|Arş.|Öğr.|Gör.|Çevirmen:|Editor:|Derleyici:/g, '');
+}
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("book");
+	
+	item.title = ZU.trimInternal(doc.querySelector('[class=product-heading]').textContent);
+
+	var authors = ZU.trimInternal(doc.querySelector('[class=manufacturers]').textContent);
+	authors = authors.split(',');
+	for (var i=0; i<authors.length; i++) {
+		var creator = cleanCreatorTitles(authors[i]);
+		item.creators.push(ZU.cleanAuthor(creator, "author"));
+	}
+	
+	var translators = ZU.xpath(doc, '//tr[contains(., "Çevirmen")]');
+	for (var i=0; i<translators.length; i++) {
+		var creator = cleanCreatorTitles(translators[i].textContent);
+		item.creators.push(ZU.cleanAuthor(creator, "translator"));
+	}
+	
+	var editors = ZU.xpath(doc, '//tr[contains(., "Editor")]|//tr[contains(., "Derleyici")]');
+	for (var i=0; i<editors.length; i++) {
+		var creator = cleanCreatorTitles(editors[i].textContent);
+		item.creators.push(ZU.cleanAuthor(creator, "editor"));
+	}
+	
+	var edition = ZU.trimInternal(doc.querySelector('[itemprop=bookEdition]').textContent);
```

Comment:
I'm a bit worried this is fragile to the absence of an edition. `ZU.trimInternal` and `textContent` will both throw an error when used on null. I'd do this in two steps:
```
var edition = doc.querySelector('[itemprop=bookEdition]');
if (edition) edition = ZU.trimInternal(edition.textContent);
```
same for all the following fields. Or you could write a function that does this and returns either the trimemd text or `null`. Setting an item element to null is fine.